### DISCUSSION TITLE
MIB2 USB connection

### DISCUSSION THREAD

**Discussion by BouvierAudi** [Mr-MIBonk/M.I.B._More-Incredible-Bash#64] [Q&A]
I appreciate the work done by everyone here, especially Harman-f. 
My car is an Audi A4 B9 2017. After purchasing I realised that it did not have Car Play or Android Auto. The only USB "port" is for charging only and is not connected to the head unit.
I upgraded the firmware from MHI2Q_ER_AUG22_P3638 to MHI2Q_ER_AUG22_P5092 (MU: 1329).
I have loaded More-Incredible-Bash and this enabled me to get the Audi Smartphone option to appear in the main menu for the first time.

I pulled out the head unit and saw that it has one unused socket which I assumed was the one which can be used for USB. See the yellow one in the photo:

![20221114_142631 (1)](https://user-images.githubusercontent.com/118609935/202862828-3e4098aa-7117-4fcc-ab01-72fa0871e0e2.jpg)

I purchased a lead that attaches to this connector and has a female USB at the other end. This was actually designed for a VW Golf but I can see no reason why it should not do the same on an Audi.

<img width="525" alt="2022-11-19" src="https://user-images.githubusercontent.com/118609935/202863283-90939cb1-5d93-4f0e-b81b-9b8194fe3957.png">

I have the USB option on the menu as a source for media but it is dimmed out even when I connect a USB stick with media files on it. So it seems that this new USB connection is not working. It does not even have power to charge a phone. I consider the following reasons:
1. The yellow socket on the head unit is not hardware configured for USB.
2. The head unit needs a software/firmware update to enable the USB to work.
3. I have got this all wrong and need to do something completely different to get a working USB connection.

I will be grateful for any guidance on what I can do before giving up on Car Play.

**Comment by harman-f** [comment]
That cable does not work with MHI2 units.
You also need to connect power from MMI and a special USB port to be able to use CarPlay.
Just AA and D-Link can be done with cheaper cables.

https://mibwiki.one/doc/usb-port-carplay-android-auto-GOvU5Ji0LN
Ali: https://de.aliexpress.com/item/4000891133662.html?gatewayAdapt=glo2deu

![image](https://user-images.githubusercontent.com/98130152/220473610-ef6fcd14-932a-46b1-b464-5ea33dd757b1.png)

**Reply by BouvierAudi** [reply]
Thanks for replying.
So the MHI2 unit does not supply power to the USB socket, so the power needs to be supplied externally. Is that correct?
I actually intend to use just Android Auto so do not need the special port for CarPlay but I assume that I do need a cable/port which connects to an external 5V power supply. 
My plan was to keep the USB socket in the glove box and connect a wireless Android Auto adapter to it. Does anyone know where I can get the 5V feed?

**Reply by harman-f** [reply]
The 5V is supplied by the green connecter, it will plug into the Back of the MHI2 QuadLock.

**Comment by harman-f** [comment]
This one will work for AA only:
https://mibwiki.one/doc/usb-port-d-link-android-auto-only-aGZ7ofACzq
Power is also provided by the MMI unit, you need to add extra wires (included in above solution)
You have to get creative to add it to the USB port in console.

Or get an AA wireless adapter and hide both somewhere behind the console.

**Reply by BouvierAudi** [reply]
I am going the route of an AA wireless adapter. I had already ordered one which will be delivered soon.
Is there anything I need to do in the SW or FW (for example in the Green Menu or using OBDEleven) to enable the USB.

**Comment by harman-f** [comment]
If you patches with M.I.B all is activated. 

Could you leave some pictures and rough instructions about your cheap AA wireless mod?

could move it to Wiki afterwards.

**Reply by BouvierAudi** [reply]
I will be very happy to provide some pictures and some notes if I get this to work. 
Currently the USB port, using the cable I showed in the photo above, is not working. I have made up a temporary splitter cable which connects the red and black wires to a USB plug which I have put into the car's "Charge only" USB socket. This at least puts power on the USB socket that would connect to a phone or wireless adapter.  But using this setup, the MMI is not even able to read a  USB stick with media files on it.

**Reply by harman-f** [reply]
Share your M.I.B log and a picture of your setup. 

Do you still try to use the cable you shared the picture? That internal wording is not correct.

**Reply by BouvierAudi** [reply]
"Do you still try to use the cable you shared the picture?"  Yes, that is the only cable I have at the moment. My knowledge of computers is limited but I am trying to learn - can you explain why this cable will not work. I thought that USB is basically 4 wires - red and black for power;  white and green for data.
I also did a scan and have fault code U11B800 "Interface for external media - no communication".

**Reply by harman-f** [reply]
That Kabel is for older units. Internal wireing is different.

**Comment by bogdandjukic** [comment]
Hello guys,
I see there is no update on this so far.

I will go this route as I had the same idea and was happy when I found this thread.
I ordered the simple cheap cable to see if it will work at all, so I'll start from there.

I'll try to document my process and post updates here.

**Reply by harman-f** [reply]
That cable is working - not for CP - have tested thwm a few times.

**Reply by bogdandjukic** [reply]
Awesome news then. The idea is to connect it to the back of the unit and leave it in the glove box. If all good, then I would get the wireless dongle for it. Just have one question. If I have the AA only cable and get the dongle that is both AA and CP, will the AA work over it? Looking at Carlinkit 4.0

**Reply by bogdandjukic** [reply]
Just to update this, maybe someone stumbles upon this thread in the future.
The **Carlinkit 4.0** connects to the car as if it is a CarPlay device, so it won't work with the cheap Android Auto only cable. At least it should not. So I guess a good device to go with would be an Android Auto only wireless device. I'll research those a bit when I get the time.

**Comment by bogdandjukic** [comment]
Ok, so the cable arrived but I couldn't make it work.
First, I connected it to the yellow connector at the back of the headunit. For the green connector that should feed the power, I already had one connected, which is the one that goes to the armrest "charge only" port.
The main issue is that when I remove that and "disassemble" the connector that goes with the cable, the now smaller 16 pn connector doesn't snap inside the quadlock connector. I managed to connect it anyway, but the phone wouldn't go into AA.

It did recognise as if it was connected to a PC (turned the debugging mode), but even when set to file transfer nothing happened.
I'll give it another go when I get some more time.

I do have a question now for the other cable, the CarPlay enabled one. I noticed that for power it only has 2 "pins", at least from the image on aliexpress.
Is there any diagram online, anywhere as to where to add those 2 pins. I guess that one is easier to connect anyway.

**Reply by harman-f** [reply]
some pictures would help, especially what you tried on the quadlock.

**Comment by bogdandjukic** [comment]
Hi guys. It's been a long time. I just want to update the thread.
I got the other connector, that supports both AA and CP. Connected it and all works as supposed.
As the connector took place of the existing one, the connectors under the armrest are not working anymore. I guess the wires could be merged so both work, but I didn't do it.
I passed the cable to the glove department, so there is where the phone sits. I ordered the wireless adapter so it should be a way better experience when it arrives.

**Reply by wiseminosse** [reply]
Can you please link the correct cable to buy?

**Reply by danieldjupvik** [reply]
Yes link please 🙏

**Reply by bogdandjukic** [reply]
Here it is. It's even on sale now for about €28 for the cable + jack:
https://www.aliexpress.com/item/1005004354452383.html?spm=a2g0o.order_list.order_list_main.51.638d1802QK2VbF

I got this combo:
![Screenshot 2024-03-25 at 10 27 43](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/8142988/d1abb946-3135-4231-abae-c6c701297e37)

**Reply by wiseminosse** [reply]
In the end I used this cable 
https://www.aliexpress.com/item/1005004354452383.html
I just had to open the green power connector to reveal the black interior and be able to adapt it to the quad lock (to open the quad lock see https://youtu.be/0xWDxiXzvKc?si=5Zc0weMxMrWtLp-R)
Worked like a charm with Android Auto. The only thing to pay attention is to code USB_FULL and not  USB_IPOD in MIB configuration.
USB_IPOD is only needed for CarPlay and requires the other (more expensive) cable

**Reply by danieldjupvik** [reply]
@wiseminosse So you need another cable than you linked for CarPlay support? Regarding the quadlock I was thinking of just tapping the already 5V wire that is used for the other "charge only" usb port. Where do you need to code the USB_FULL or USB_IPOD? Is it with OBDeleven because I've already enabled CarPlay with the More Incredible Bash patch, but more needs to be coded?

**Reply by wiseminosse** [reply]
For CarPlay you need the other cable with adapter as documented in the other thread (https://www.aliexpress.com/item/1005004354452383.html)

The coding steps
5F Multimedia
Adaptation
Car Configuration or Vehicle configuration
Google_GAL -> ON
Apple_DIO -> ON
Mirror_link -> ON
Smartphone_interface_for_rhmi -> ON

Long coding
byte_19_USB -> USB_ipad [if you use the most expensive cable - required for CarPlay]
byte_19_USB -> USB_full [if you use the cheaper cable - working only for Android Auto]

**Comment by Moisesng86** [comment]
Hello,

In the last week I have made the update on my car.... and it Works Perfectly :). I will describe here what I have done in case it can help.

### Initial Problem
- My car only has one USB port on the central console for "Charge only"
- Android Auto is already activated but I could not connect since no USB Data isavailable
- I do not want to loss the USB connector in the central console nor the Induction charge

### Idea
- I wanted to place a USB connector inside the Glove box
![USB_place](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/155737831/d41f1787-698d-419f-843a-0081b9d43480)

### Needed Material
- Cable and USB connector: [Cable + USB Connector](https://a.aliexpress.com/_EjIipWt )
- Branch Connector (this is used to connect - in parallel - the USB 5V to the original cable. So I do not lose the current USB Charge Only port): [Branch Connectors](https://www.amazon.de/dp/B09WY5MXFX?ref=ppx_pop_mob_ap_share)
- Wireless Adapter: It is optional. Since I am placing the USB inside the glove box it is quite useful: [CarlinKit 5.0](https://www.amazon.de/dp/B0C149F6CG?ref=ppx_pop_mob_ap_share)
- [Cloth Hanger](https://th.bing.com/th/id/OIP.pWc2EdfugA4YC93KC0mMYQAAAA?rs=1&pid=ImgDetMain): It will help to pass the cable through without dismounting the glove box
- [Shoelaces](https://en.wikipedia.org/wiki/Shoelaces): One we pass the Cloth Hanger, this will help also.
- [Cutter](https://th.bing.com/th/id/R.7271969468c737f8cf98640ea8808bec?rik=sPjTxtKaeKrapw&riu=http%3a%2f%2fupload.wikimedia.org%2fwikipedia%2fcommons%2fc%2fcf%2fBox-cutter.jpg&ehk=OwBVB3IEWtf4oJTtp3mqaP46LCYKJShHmQxmOZfdPAM%3d&risl=&pid=ImgRaw&r=0)

### Process
1. With the cutter, cut the 3 "notches" that keep together the backside with the frame. Then you will be able to move it a bit back and pass the cable
![cuts](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/155737831/e82ea13a-3658-4c99-ad11-354a3bb633ca)

2. Take out the ratio and disconnect the quadlock
3. With a lot of love and patient, try to pass the Cloth Hanger from the small hole (behind the radio) to the USB compartment that we have already prepare (step 1.).
4. For me It was really difficult to tie the cable to the cloth hanger, I first tie the shoelace to the hanger and I pass it. So at the end, I have the shoelace between the radio compartment and the USB compartment.
5. Pass the cable from the radio compartment to the USB compartment using the shoelace as a guide.
6. Connect the cable Yellow connector to the radio Yellow connector
![connector](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/155737831/d5b17ec9-470b-4a82-8e74-70079414666a)

7. Using the Quadlock pin layout [Pin Layout](https://mibwiki.one/doc/quadlock-pin-layout-u5Qkyar282) connect USB the 5V and Ground cables to the original cables using the Branch Connector. I have used the blue ones, the smaller ones. 
![parallel](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/155737831/ee612c30-b922-4b5e-a906-70ed5c1ba2b6)

8. Place the radio back
9. Connect the Wireless Adapter (optional)

### Result
![Final](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/155737831/710d9576-baf8-42e6-8091-9d2ca2b61419)

![Final zoom](https://github.com/Mr-MIBonk/M.I.B._More-Incredible-Bash/assets/155737831/ba26137c-9c31-42c8-a82f-bd38f09493d0)


Hope this helps!!

**Reply by Barbsterlive** [reply]
Is this working on AA or CP? you didnt specified it, 

i've the same idea of you but in the previous replies i understood that some of the cable mentioned are good only for AA and i didnt get exactly which one

Ty

**Comment by Walidrasuly** [comment]
Hello everyone
I have a question about my vw Sharan 2016
My carplay only supports Android, I have a USB(1Q0 035 346 A)
Does anyone have an id with which model usb port I can replace the iPod suppoted ?
thenks

**Comment by NoIdeaWhutIDo** [comment]
Hello, 
Maybe someone had experience why Carplay sometimes pops up after pluging iPhone in usb, but most of the time it only charge the phone without any notification on MMI screen. Audi A4 patched with this wire set up and connector. 
All ideas and comments are welcome.

https://www.aliexpress.com/item/1005004354452383.html?spm=a2g0n.order_detail.order_detail_item.3.2cbcf19cpuh1mX&_gl=1*uojmxv*_gcl_aw*R0NMLjE3MzI5MTE0ODQuQ2p3S0NBaUE2YVc2QmhCcUVpd0E2S3pEYzBxcXNCTTJmbU1jU1o3RkwwUnJLLVZrZVZENVRJcVRNZF9kN3Y5RjdGOG55RGFYMmY0cW94b0N2OGNRQXZEX0J3RQ..*_gcl_dc*R0NMLjE3MzI5MTEzOTYuQ2p3S0NBaUE2YVc2QmhCcUVpd0E2S3pEY19BUWgzMDc4dW5MaTVJd2RsTFNSMUZUQjVwcHhkemtrbjFOQTdjemlmdXVVQXprRzVqd2lob0NGQzBRQXZEX0J3RQ..*_gcl_au*MTYwNjQwMDE5Mi4xNzMyNTU1MzI4*_ga*Njc5NjgzNTg2LjE3MzExNDMzNzc.*_ga_VED1YSGNC7*MTczNDA2OTQ5OC4xMjQuMS4xNzM0MDcwNDYzLjU3LjAuMA..

**Reply by Cuco77** [reply]
Buonasera, sto seguendo questa discussione ma essendo italiano leggo i commenti tradotti. Vorrei installare la USB nella mia auto ma non mi è ancora chiaro se il cavo+USB economico è compatibile con Android auto o no. Grazie .

**Comment by Walidrasuly** [comment]
https://youtu.be/7yTYaxzeXfY?si=iOIWDMnEuR8J97ZD

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by Cuco77** [reply]
![Screenshot_2024-12-15-21-25-17-071_com android chrome](https://github.com/user-attachments/assets/9957899d-93e0-40e3-9767-65e37bfda005)
Questo cavo non funziona?

**Reply by tox-stack** [reply]
> ![Screenshot_2024-12-15-21-25-17-071_com android chrome](https://private-user-images.githubusercontent.com/186136125/395893632-9957899d-93e0-40e3-9767-65e37bfda005.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzY4ODk2ODgsIm5iZiI6MTc3Njg4OTM4OCwicGF0aCI6Ii8xODYxMzYxMjUvMzk1ODkzNjMyLTk5NTc4OTlkLTkzZTAtNDBlMy05NzY3LTY1ZTM3YmZkYTAwNS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQyMlQyMDIzMDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNzYwYTI2MDQ5N2U4OGIwNWVkMjI3YmI2NmZkYTkwNzY3NGM0ZGEzMTRkZDM0OWU5MzdkMjdlMzFhY2Q5OTQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZqcGVnIn0.B7fDYSDEB2bNOQKZiVpvndzNICY04kJDQUoCaWkDXXI) Questo cavo non funziona?

Yes it works.

**Comment by Walidrasuly** [comment]
about which car and year of manufacture


Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by Cuco77** [reply]
Audi Q2 del 2017 con sistema Mib2 . Ho già attivato sbloccato il sistema. Mi manca solo una nuova USB. Uso solo android.

**Comment by Walidrasuly** [comment]
https://youtu.be/WgCl5UoqI_M?si=41oN1TxakP9eEwkx
YouTube<https://youtu.be/WgCl5UoqI_M?si=41oN1TxakP9eEwkx>
Share your videos with friends, family, and the world
youtu.be


Verzonden vanuit Outlook<http://aka.ms/weboutlook>
________________________________
Van: Cuco77 ***@***.***>
Verzonden: zondag 15 december 2024 21:34
Aan: Mr-MIBonk/M.I.B._More-Incredible-Bash ***@***.***>
CC: Walidrasuly ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [Mr-MIBonk/M.I.B._More-Incredible-Bash] MIB2 USB connection (Discussion #64)


Audi Q2 del 2017 con sistema Mib2

**Comment by pablosguajardo** [comment]
Hi, I made a connection to test with a simple USB cable and it works perfectly.
Here's an image showing how to test it. Or, if you have an incompatible cable, how to fix it.
It was tested with the VW NIB REGIO model (models such as the Nivus, T-Cross, and Virtus).
![image](https://github.com/user-attachments/assets/c7851b9a-9432-4628-a44a-a0af39fc4cd4)

![image](https://github.com/user-attachments/assets/7c4f8148-fd96-4c15-b241-bc3399bca91a)

![image](https://github.com/user-attachments/assets/825b6f0b-85f3-4f3e-b776-f9d55c15ee99)

**Comment by pitone81** [comment]
Hi everyone, I followed the guide and installed this (https://a.aliexpress.com/_EjIipWt). I have a small problem when I connect my phone. Android Auto opens and closes immediately. After I try a couple of times, it opens, but stays open for 20 seconds, and then boom, it closes again. I tried Apple and everything works. The USB port is set to USB_iPod.

**Reply by tox-stack** [reply]
Set to usb_full
